src/xenia/cpu/hir/value.cc

```diff
@@ -750,6 +750,12 @@ void Value::Not() {
   }
 }
 
+void Value::AndNot(Value* other) {
+  assert_true(type == other->type);
+  other->Not();
```

Comment:
This will modify the `other` passed to the function with an effect on the caller — is this what is expected from this function? I think we should create a local copy of `other` for this purpose if that's possible.

Reply:
I also think that for this purpose, `other` should be `const` in every function where `other` is not expected to be modified.

Reply:
I agree! To be perfectly clean we should use local copy then on it execute Not()

```
Value* second = new Value(*other);
second->Not();
And(second);
```

About constness. Sure but in different PR maybe?

Reply:
In the example you've provided, `second` will never be destroyed.